packages/gitbook/src/components/TableOfContents/useScrollToActiveTOCItem.tsx

```diff
@@ -0,0 +1,55 @@
+'use client';
+
+import React from 'react';
+
+const TOCScrollContainerContext = React.createContext<string | null>(null);
+
+const SCROLL_OFFSET = 200;
+
+/**
+ * Provider that supplies the ID of the scroll container for the table of contents.
+ */
+export function TOCScrollContainerProvider(
```

Comment:
Instead of just rendering the react context, you can extend it to: create a ref, render the div, render the contxt provider with the ref as value.

It'll avoid the use of an ID for it.

Reply:
Yeah I had tried with passing a ref via the provider but that would not work because of the TableOfContents component is server side. But maybe because I was creating it directly in the TOC component instead of doing it in the Provider itself. I'll try what you suggested.